src/FragmentMakers.cpp

```diff
@@ -0,0 +1,80 @@
+#include "trigger/FragmentMakers.hpp"
+#include "dataformats/trigger/TriggerPrimitivesFragment.hpp"
+#include "triggeralgs/TriggerPrimitive.hpp"
+
+namespace dunedaq::trigger {
+
+//======================================================================
+std::unique_ptr<dataformats::Fragment>
+make_fragment(std::vector<TPSet>& tpsets)
+{
+  std::vector<triggeralgs::TriggerPrimitive> tps;
+  for (auto const& tpset : tpsets) {
+    for (auto const& tp : tpset.objects) {
+      tps.push_back(tp);
+    }
+  }
+
+  return make_fragment(tps);
+}
+
+//======================================================================
+std::unique_ptr<dataformats::Fragment>
+make_fragment(std::vector<triggeralgs::TriggerPrimitive>& tps)
+{
+  size_t n_bytes = sizeof(dataformats::TriggerPrimitivesFragment) +
+                   tps.size() * sizeof(dataformats::TriggerPrimitivesFragment::TriggerPrimitive);
+  std::unique_ptr<uint8_t[]> buffer(new uint8_t[n_bytes]);
+  dataformats::TriggerPrimitivesFragment* tpf = reinterpret_cast<dataformats::TriggerPrimitivesFragment*>(buffer.get());
+  tpf->version = dataformats::TriggerPrimitivesFragment::s_tpf_version;
+  tpf->n_trigger_primitives = tps.size();
+  size_t counter = 0;
+  for (auto const& tp : tps) {
+    dataformats::TriggerPrimitivesFragment::TriggerPrimitive& tp_fragment = tpf->primitives[counter++];
```

Comment:
fragment_tp might be a better name, I thought for a moment that this was a TriggerPrimitivesFragment object.

Reply:
Yes, that's a better name. Done in 5eade9a